recipes/logfault/all/conanfile.py

```diff
@@ -0,0 +1,135 @@
+    settings = "os", "arch", "compiler", "build_type"
+    # Do not copy sources to build folder for header only projects, unless you need to apply patches
+    no_copy_source = True
+    generators = "CMakeToolchain", "CMakeDeps"
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    # In case the project requires C++14/17/20/... the minimum compiler version should be listed
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "15",
+            "clang": "15",
+            "gcc": "13",
+            "msvc": "14.31",
+            "Visual Studio": "17.1",
+        }
+    
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables['LOGFAULT_BUILD_TESTS'] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+
+    # Use the export_sources(self) method instead of the exports_sources attribute.
+    # This allows finer grain exportation of patches per version
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        # src_folder must use the same source folder name than the project
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        # Prefer self.requires method instead of requires attribute
+        # Direct dependencies of header only libs are always transitive since they are included in public headers
+        #self.requires("dependency/0.8.1", transitive_headers=True)
+        if not self.conf.get("tools.build:skip_test", default=False):
+            self.requires("gtest/1.14.0")
+
+    # same package ID for any package
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            # Validate the minimum cpp standard supported when installing the package. For C++ projects only
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        # In case this library does not work in some another configuration, it should be validated here too
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(f"{self.ref} can not be used on Windows.")
+
+    def source(self):
+        # Download source package and extract to source folder
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    # Not mandatory when there is no patch, but will suppress warning message about missing build() method
+    def build(self):
+        # The attribute no_copy_source should not be used when applying patches in build
+        apply_conandata_patches(self)
```

Comment:
```suggestion
    def build(self):
```

Reply:
Thank you for the review!

I will go over it in detail and push a new revision later today (I'm in Europe).